### ISSUE TITLE
Log option for UniformFloatHyperparameter

### ISSUE THREAD

**Issue by bmmalone** [automl/ConfigSpace#60]
Hi,

This is more of a question than an issue; sorry if this is the wrong place to post.

I have been working on extending auto-sklearn to include other types of components, and several of these have regularization penalty-esque hyperparameters. For similar types of hyperparameters (e.g., alpha and the thetas for Gaussian processes), the `log` option is always set to `True` when creating the respective `ConfigSpace` hyperparameters. 

Does this result in the optimizer using a type of logarithmic search space for the respective hyperparameter (e.g., I just evaluated d=0.11 now consider something like d=0.011 or d=1.1 rather than d=0.12)? Also, are there better/other ways to specify these types of hyperparameters?

Have a good day,
Brandon

**Comment by mfeurer** [comment]
The argument `log` results in the hyperparameter being sampled and model in log space. More specific, let's call the hyperparameter value `v`. You could also write `v` as `2^x` or `10^x` and then model `x` instead. This is helpful for parameters like learning rate, where the change in behavior when going from `0.01` to `0.1` is much more drastic than when going from `0.01` to `0.02`'

Whether it makes sense for the Gaussian process hyperparameters is a good question. I can't really answer that because I'm not very familiar with the Gaussian process in scikit-learn.

**Comment by bmmalone** [comment]
Thanks for your quick clarification, Matthias.

Just for reference, [GPs in auto-sklearn](https://github.com/automl/auto-sklearn/blob/master/autosklearn/pipeline/components/regression/gaussian_process.py) do use the log space representations; some of our new components have hyperparameters with similar properties, so that was what prompted checking.
